# HG changeset patch
# User Rik <rik@octave.org>
# Date 1380906044 25200
#      Fri Oct 04 10:00:44 2013 -0700
# Node ID 4166048ba6cfbb9cad08b4b6e53bfd8d80d6fae7
# Parent  ddecb2e231d473eac16d9fcf8fe28ed195908797
text.m: Fix text location if "Position" specified as prop/value pair.

* scripts/plot/text.m: Search for "position" property in input PROP/VAL pairs
and use coordinates if available.

diff --git a/scripts/plot/text.m b/scripts/plot/text.m
--- a/scripts/plot/text.m
+++ b/scripts/plot/text.m
@@ -126,21 +126,30 @@ function h = text (varargin)
 
   ## Get axis argument which may be in a 'parent' PROP/VAL pair
   [hax, varargin] = __plt_get_axis_arg__ ("text", varargin{:});
 
   if (isempty (hax))
     hax = gca ();
   endif
 
+  ## Position argument may alse be in PROP/VAL pair
+  idx = find (strcmpi (varargin, "position"), 1);
+  if (idx)
+    pos = varargin{idx+1};
+    varargin(idx:idx+1) = [];
+  else
+    pos = [x(:), y(:), z(:)];
+  endif
+
   if (nx == ny && nx == nz && (nt == nx || nt == 1 || nx == 1))
-    pos = [x(:), y(:), z(:)];
     htmp = zeros (nt, 1);
     if (nx == 1)
       htmp = __go_text__ (hax, "string", string{1},
+                               ## varargin first, in case "Units" set for pos.
                                varargin{:},
                                "position", pos);
     elseif (nx == nt)
       for n = 1:nt
         htmp(n) = __go_text__ (hax, "string", string{n},
                                     varargin{:},
                                     "position", pos(n,:));
       endfor
